File: Python/CreateAndDeleteNodes.py
from time import sleep
import traceback
import random
from LinkedList import LinkedList
from Node import Node


class CreateAndDeleteNodes():
    def __init__(self, initsize=0, listsize=0, inserts=0, delaymilisec=0, rate=0):
        self.initsize = initsize
        self.listsize = listsize
        self.inserts = inserts
        self.delaymilisec = delaymilisec
        self.rate = rate

    def runtest(self):
        mylist = LinkedList.initialize()

        for i in range(self.initsize):
            tempint = random.randint(1, 2147483647)
            tempnode = Node.initializewithargs(tempint)
            mylist.insertatbeginning(tempnode)

        for i in range(self.inserts):
            for i in range(self.rate):
                sleep(self.delaymilisec)
                tempint = random.randint(1, 2147483647)
                tempnode = Node.initializewithargs(tempint)
                mylist.insertatend(tempnode)
            sleep(self.delaymilisec)
            mylist.removefirst()


test = CreateAndDeleteNodes(10000, 100, 10000, .001, 2)
test.runtest()

# print(
#    str(test.listsize) + ' ' +
#    str(test.initsize) + ' ' +
#    str(test.inserts) + ' ' +
#    str(test.delaymilisec) + ' ' +
#    str(test.rate)
# )

File: Java/CreateAndDeleteNodes.java
import java.util.*;

public class CreateAndDeleteNodes {

    public static void main(String[] args){

        CreateAndDeleteNodes createAndDeleteNodes = new CreateAndDeleteNodes();
        createAndDeleteNodes.createAndDeleteNodes(10000, 10000, 1, 2);

    }

    /* creates and deletes nodes from linked list
      * initSize: size list is to be initially populated to
      * inserts: number of nodes to be inserted after initial array is created
      * delayMiliSec: delay time between insertions and deletions (for VisualGC)
      * rate: creation to deletion ratio (e.g. 2 -> two inserts for every one deletion)
    */
    void createAndDeleteNodes(int initSize, int inserts, int delayMiliSec, int rate){
        
        Random rand = new Random();
        LinkedList myList = new LinkedList();
        int tempInt;
        
        // populate list to initial size
        for(int i = 0; i < initSize; i++){
            tempInt = rand.nextInt(2147483647);
            Node temp = new Node(tempInt);
            myList.insertAtBeginning(temp); 
        }

        // insert nodes at end while deleting nodes at beginning
        for(int i = 0; i < inserts; i++){
            for(int j = 0; j < rate; j++){
                sleep(delayMiliSec);
                tempInt = rand.nextInt(2147483647);
                Node temp = new Node(tempInt);
                myList.insertAtEnd(temp);
            }
            sleep(delayMiliSec);
            myList.removeFirst();
        }

    }
    
    void sleep(int miliseconds){

        try {
            Thread.sleep(miliseconds);
        } catch (InterruptedException e){
            e.printStackTrace();
        }
    
    }

}
